### PR TITLE
feat($md-contact-chips): add md-chips feats to md-contact-chips

### LINKED ISSUE #4537: md-contact-chips readonly
Hi,

Could you please add readonly to md-contact-chips, or make ng-disable working with it.

Thanks

### PR DESCRIPTION
**Pull request by apat183**
Added md-chips features (readonly, md-on-add, md-on-remove, and md-on-select) to md-contact-chips by binding and passing expressions from the md-contact-chips directive to the underlying md-chip.

Closes #4537.

### CODE REVIEW

File: src/components/chips/js/contactChipsDirective.js [hunk 1]
```diff
@@ -100,35 +108,39 @@ var MD_CONTACT_CHIPS_TEMPLATE = '\
...
+        template: function(element, attrs) {
+            return MD_CONTACT_CHIPS_TEMPLATE;
+        },
+        restrict: 'E',
+        controller: 'MdContactChipsCtrl',
+        controllerAs: '$mdContactChipsCtrl',
+        bindToController: true,
+        compile: compile,
+        scope: {
+            contactQuery: '&mdContacts',
```
Comment (DevVersion) [inline src/components/chips/js/contactChipsDirective.js]: You should indent with two spaces.

Comment (apat183) [inline src/components/chips/js/contactChipsDirective.js]: Apologies for that, my editor kept adding the indents.

**Comment by kpgarrod** [comment]
I really need this feature. Will it be merged in the next release please?

**Comment by apat183** [comment]
If you really need this for now use this temp solution: http://codepen.io/anon/pen/YwjzGp 

It would be easy to replace once this has been merged. I'm sure there is more work to be done before it could be merge.

**Comment by kpgarrod** [comment]
Hi Anand,

Yes, I do need it urgently, so thanks for the temp solution!



On Fri, Feb 5, 2016 at 1:59 PM Anand Patel <notifications@github.com> wrote:

> If you really need this for now use this temp solution:
> http://codepen.io/anon/pen/YwjzGp
>
> It would be easy to replace once this has been merged. I'm sure there is
> more work to be done before it could be merge.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/6961#issuecomment-180344967>.
>
-- 
Keith Garrod
Phone +356 2156 9774
Mobile +356 9958 6774

**Comment by apat183** [comment]
If you need access to the object for md-on-add or md-on-remove in the example I provided, just pass $chip in the function. e.g. md-on-add="yourFunction($schip)"

**Comment by topherfangio** [comment]
@apat183 Just came across this PR when working on some other chip issues. Do you have the time to rebase this against master, undo the extra spacing that was added, and maybe add the following new attributes as well?

* `md-transform-chip`
* `md-separator-keys`
* `delete-hint`
* and `delete-button-label`

Thanks!

**Comment by apat183** [comment]
Yep I can try and do that.

**Comment by topherfangio** [comment]
@apat183 Just noticed that you added some extra commits. Is this ready for review?

**Comment by apat183** [comment]
@topherfangio no it still needs more work. The last time I pushed the commit `delete-hint` and `delete-button-label` had layout issues. `md-max-chips` was working though. Is it possible for someone else to pick up from the initial commit?

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Pull request closed without merging**